src/scoring_functions.py

```diff
@@ -23,7 +23,8 @@ def cosine_sim(predicted_user_profiles, predicted_item_attributes):
     cosine_similarities = predicted_scores / norms
     re_ranked_scores = predicted_scores - alpha * cosine_similarities
     # add minimum value of each row to all item scores to ensure scores are positive.
-    re_ranked_scores += np.abs(np.min(re_ranked_scores, axis=1))[:, np.newaxis]
+    # re_ranked_scores += np.abs(np.min(re_ranked_scores, axis=1))[:, np.newaxis]
+    re_ranked_scores = np.exp(re_ranked_scores)
```

Comment:
are we omitting content fairness? just double-checking

Reply:
this is cosine_sim. Content fairness I am trying to fix